PROJECT_GUIDELINES.md

```diff
@@ -0,0 +1,228 @@
+- Tell us how to reproduce the issue (eg. is it happening in a specific browser?)
+- Can it be reliably reproduced? If not, tell us how often it happens and under what circumstances.
+- Screenshots and gifs are our friends!
+- Did this problem start happening after a recent release or was it always a bug?
+
+## Code base
+
+Our code base is written in TypeScript. We love it because it keeps our code type safe. Everything should have a type and `any` should be avoided whenever possible.
+
+1. Please follow [tslint-iconic-rules](https://github.com/ionic-team/tslint-ionic-rules/blob/master/tslint.js).
+
+2. Variable decorators should be inlined.
+   `@Prop() name: string;`
+
+3. Method decorators should be multi-line.
+
+```
+@Listen('click')
+onClick() {
+ ...
+}
+```
+
+4. Use private variables and methods when possible. These are useful for detecting deadcode and enforcing encapsulation. Note that this is a feature which TypeScript provides to help harden your code, but using `private`, `public` or `protected` does not make a difference in the actual JavaScript output.
+
+5. Code with Method/Prop/Event/Component decorators should have jsdocs. This allows for documentation generation and for better user experience in an editor that has TypeScript intellisense.
+
+### A11y
+
+Components should be accessible and we use the following resources as guides:
+
+- [Google accessibility overview](https://developers.google.com/web/fundamentals/accessibility/)
+- [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)
+
+### Components
+
+Before the checklist, new components should have an [issue](https://github.com/Esri/calcite-app-components/issues/new?assignees=&labels=new+component%2C+0+-+new%2C+architecture&template=new-component.md&title=New+Component%3A+) so we can determine whether they belong here or calcite-components. If a component does belong here, that's where the [checklist](https://github.com/Esri/calcite-app-components/wiki/New-Component-Checklist) can be used to guide development.
+
+#### General checklist
+
+- Slots are documented at the beginning of each component file after the import statements and before the `@Component` decorator.
+- Please properly mutate and reflect props.
+- Make sure to document props.
+- UI text props start with `text`. For example: `textCollapse` updates the label of the collapse icon when the component is expanded in the calcite action bar.
+- Events are named in present tense and start with the `calciteComponent`. For example: `calciteBlockToggle` is emitted when the header in a calcite block has been clicked.
+- Properties without values are marked as required (with a `!` in front) or optional (`?`). For example: `@Prop() label?: string;` or `@Prop() text!: string;`
+- Set `shadow: true` in the `@Component` options to make sure styles are encapsulated.
+- Naming conventions of properties, methods, and events are consistent between components for consistency.
```

Comment:
Maybe rephrase this to: `For consistent naming, look at other components to see if there's an existing property/method/event name to follow.`